jormungandr/src/fragment/pool.rs

```diff
@@ -98,11 +98,15 @@ impl Pools {
         for fragment in fragments.by_ref() {
             let id = fragment.id();
 
+            let _enter =
+                tracing::span!(tracing::Level::TRACE, "pool_incoming_fragment", fragment_id=?id);
```

Comment:
I think the `enter` guard is returned only if you call `span!(...).enter()`, otherwise the span is not active

Reply:
Could use `trace_span!` macro for brevity.

Reply:
uhm, take care because iirc enter do not work properly with async. I'll come back with some info.

Reply:
﻿﻿>  ⚠ ️Warning
> Warning: In asynchronous code that uses async/await syntax, Span::enter may produce incorrect traces if  the returned drop guard is held across an await point. See the method documentation for details.

Reply:
Although it seems Ok to use here. But just to be sure.